### PR TITLE
[coqide] fix procedure to parse arguments

### PR DESCRIPTION
**Pull request by gares**
coqide calls coqidetop -batch, if you are in -async-proof on then
coqidetop spawns a worker and passes -batch to it. At some point,
I could not find the commit, this made the worker die.

### CODE REVIEW

**Comment by gares** [comment]
On linux it seems it works anyway, but on windows this death is
perceived by coqide which then does not start.

**Comment by gares** [comment]
The error can be see on linux as well by `coqidetop -nois -batch -async-proofs on a.v`, with this patch you see no such an error.

**Comment by gares** [comment]
I've tested the fix on a windows vm, it works.

**Comment by ejgallego** [comment]
It seems to me that the right fix is to actually not use `-batch` (which by the way it is ignored) , but `-quiet`.

**Comment by ejgallego** [comment]
There is something very bizarre here , Ummmm.

**Comment by ejgallego** [comment]
Yeah so actually coqide is expecting `-batch` to actually terminate the `coqidetop` process, what a mess. I guess this is fine then.

File: toplevel/workerLoop.ml [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 let rec parse = function
   | "--xml_format=Ppcmds" :: rest -> parse rest
+  | "-batch" :: rest -> parse rest
```
Comment (ejgallego) [inline toplevel/workerLoop.ml:13]: Maybe this should be added to `stm/asynctaskqueue` ?

**Comment by gares** [comment]
I don't know how we ended up calling coqtop to know which options coqide can accept, but yes, that is the mess.

Of course it could be moved there, but here there was already another "skip this thing", so I did follow the previous code on this.
If we refactor this argument parsing mess, then I think it is better to have all the crap in the same place.

**Comment by ejgallego** [comment]
> Of course it could be moved to the stm/, but here there was already another "skip this thing", so I did follow the previous code on this. If we refactor this argument parsing mess, then I think it is better to have all the crap in the same place.

Note that we already filter the options we pass to the workers in that file, I ignore why we have this other stuff but the place I pointed out seems better no? Likely when I added that here I didn't know of the filter in stm.

But the problem is quite interesting, for some reason on Linux the workers are not being initialized on the test `-batch` calls, is it possible that some windows workaround is creating this difference? We didn't use to disable async on win:

```ocaml
let spawn_handle args respawner feedback_processor =
  let prog = coqtop_path () in
  let async_default =
    (* disable async processing by default in Windows *)
    if List.mem Sys.os_type ["Win32"; "Cygwin"] then
      "off"
    else
      "on"
  in
```

File: toplevel/workerLoop.ml (hunk 1 above)
Comment (gares) [reply toplevel/workerLoop.ml]: done

**Comment by gares** [comment]
please check the updated fix and assign

**Review by ejgallego — APPROVED** [review]

**Comment by gares** [comment]
ci gree (for the part that could matter here)

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**